core/include/msg_bus.h

```diff
@@ -50,15 +50,21 @@ typedef struct {
     kernel_pid_t pid;       /**< Subscriber PID */
 } msg_bus_entry_t;
 
+/**
+ * @brief Flag set on `sender_pid` of `msg_t` that indicates that
+ *        the message was sent over a bus.
+ */
+#define MSB_BUS_PID_FLAG    (1U << ((8 * sizeof(kernel_pid_t)) - 1))
```

Comment:
Since `kernel_pid_t` is signed, this will effectively make all bus messages use a negative PID.
I did not find any other use for negative PIDs, but if they already have a meaning assigned we might as well use

```suggestion
#define MSB_BUS_PID_FLAG    (1U << ((8 * sizeof(kernel_pid_t)) - 2))
```

instead.

Reply:
@kaspar030 should know this.